math/wide_integer/uintwide_t.h

```diff
@@ -316,21 +316,21 @@
     }
 
   protected:
-    mutable size_type elem_count;
-    pointer           elems;
+    mutable size_type elem_count; //NOLINT(cppcoreguidelines-non-private-member-variables-in-classes,misc-non-private-member-variables-in-classes)
+    pointer           elems;      //NOLINT(cppcoreguidelines-non-private-member-variables-in-classes,misc-non-private-member-variables-in-classes)
   };
 
   template<typename ValueType, typename AllocatorType>
-  bool operator==(const dynamic_array<ValueType, AllocatorType>& lhs,
-                  const dynamic_array<ValueType, AllocatorType>& rhs)
+  auto operator==(const dynamic_array<ValueType, AllocatorType>& lhs,
+                  const dynamic_array<ValueType, AllocatorType>& rhs) -> bool
   {
-    bool left_and_right_are_equal;
+    bool left_and_right_are_equal = false;
```

Comment:
Have you considered avoiding this variable altogether and simply returning the result?

Reply:
Good catch. I will take a look. Thanks John!

Reply:
Well as it turns out this is an old habit. Some old-school C-style guidelines recommend in the code metrics one and only one return point from a given function. Now I don't adhere to this in the _entire_ project. But where it is easy, sure, i limit to one single return path per function.

On top of that, this is one of the instances where `clang-tidy` erroneously claimed it was an uninitialized variable.

So... al-in-all, I'm done with that code sequence and will leave it and similar other ones alone.

Reply:
Herewith I am closing the final review point.

Many, many thanks @johnmcfarlane for an excellent review with numerous helpful review findings.